### ISSUE TITLE
Packages contain code that will be rejected by a future version of Rust

### ISSUE THREAD

**Issue by lexika979** [lemunozm/message-io#131]
Hello,
after adding message-io to new project and compiling, i got the following message:

--------------------------------------------------------------------------------------------------------------------
warning: the following packages contain code that will be rejected by a future version of Rust: ntapi v0.3.7
note:
To solve this problem, you can try the following approaches:

- Some affected dependencies have newer versions available.
You may want to consider updating them to a newer version to see if the issue has been fixed.

ntapi v0.3.7 has the following newer versions available: 0.4.0

- If the issue is not solved by updating the dependencies, a fix has to be
implemented by those dependencies. You can help with that by notifying the
maintainers of this problem (e.g. by creating a bug report) or by proposing a
fix to the maintainers (e.g. by creating a pull request):

  - ntapi@0.3.7
  - Repository: https://github.com/MSxDOS/ntapi
  - Detailed warning command: `cargo report future-incompatibilities --id 4 --package ntapi@0.3.7`

- If waiting for an upstream fix is not an option, you can use the `[patch]`
section in `Cargo.toml` to use your own version of the dependency. For more
information, see:
https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section

note: this report can be shown with `cargo report future-incompatibilities --id 4`

--------------------------------------------------------------------------------------------------------------------

It seems that just updating the dependency would fix this issue, in case it does not i would make a issue in their repo :)

**Comment by lemunozm** [comment]
Hi! I'm not able to check the error because I'm not in windows, and `ntapi` is only compiled there. I do not use this dependency directly in `message-io`, so it seems that there is any other dependency that includes that. I've updated all dependencies in this branch: `dependencies-updated`. 

Could you please check if importing the updated version into your project still generates the issue?
To address to this specific change in Cargo.toml:
```toml
message-io = { git = "https://github.com/lemunozm/message-io", branch = "dependencies-updated" }
``` 
Thanks!!

**Comment by lexika979** [comment]
Unfortunately that didnt fix it :(
I cant seem to track down the dependency in message-io that causes this, because of the limited output i am getting from cargo.
Perhaps we could use a [patch] like in message-io to get rid of this like cargo suggested? That would (rather forcefully) get rid of the outdated dependency.

**Comment by lexika979** [comment]
If thats a valid option i'd make a PR :)

**Comment by lemunozm** [comment]
To be honest I've never dealt with this problem, but if that patch works, I'm fine with the PR 😃

**Comment by lexika979** [comment]
After trying to fix this i just noticed that i cannot [patch] the version of a dependency's dependency :(
The only way to fix this is to actually track down what dependency includes ntapi and then file a PR/Issue there

**Comment by lexika979** [comment]
I will try to track down the dependency if i have some free time, but at the moment i have lots of work to do so no idea when i will get around to doing it - I'll keep you updated

**Comment by lemunozm** [comment]
Thanks for the info 😃. One question, a `cargo tree` in Windows does not show the `ntapi` dependency somewhere?

**Comment by sigmaSd** [comment]
You can use `cargo tree --target x86_64-pc-windows-gnu` on linux, the dep is coming from mio 
![image](https://user-images.githubusercontent.com/22427111/195992730-f8c00740-a64f-45f8-b55b-5bc6fd9f7fd2.png)

Mio already have replaced those deps with windows-sys in latest version https://github.com/tokio-rs/mio/pull/1556

**Comment by lemunozm** [comment]
Thanks, @sigmaSd, for the advice!
Seems like it's time to update `mio` to version 0.8. I'll try to get some time to do it.

**Comment by garin1000** [comment]
Hi @lemunozm,

I just wanted to ask when you could release a message-io version with the updated dependency :-)

All that's really needed is changing the Cargo.toml to use mio 0.8, which  I did locally. All message-io tests run fine, so does my application.

Thanks!

**Comment by lemunozm** [comment]
Done! Published version `0.14.7` with `mio` version `0.8`

**Comment by garin1000** [comment]
Thank you!! That was quick!
